eq-author/src/App/qcodes/QCodesTable/index.js

```diff
@@ -38,304 +64,314 @@ const StyledTableBody = styled(TableBody)`
+        summaryAnswers,
+      } = item;
+
+      const label = removeHtml(totalTitle);
+
+      answerRows.push({
+        title,
+        alias,
+        answers: [{ id, type, qCode, label, summaryAnswers }],
+      });
     }
   }
-  if (confirmation) {
-    const { id, title, qCode, __typename: type } = confirmation;
-    const label = "";
-
-    const confirmationRow = (
-      <Row
-        key={id}
-        id={id}
-        alias={alias}
-        title={title}
-        type={answerMatrix[type]}
-        label={label}
-        qCode={qCode}
-      />
-    );
-    rowBuilder.push(confirmationRow);
-  }
 
-  return rowBuilder;
+  return { answers: answerRows };
 };
 
-const buildContent = sections => {
-  const content = [];
-
-  for (const section of sections) {
-    const pages = section.pages;
-    const rows = pages.map(page => {
-      const rowBuilder = [];
-      const { answers, summaryAnswers } = page;
-      if (answers) {
-        const numberOfAnswers = answers.length;
-        if (numberOfAnswers) {
-          const questionRow = builtRows(page);
-          rowBuilder.push(questionRow);
-        }
-      } else if (summaryAnswers) {
-        const numberOfAnswers = summaryAnswers.length;
-        if (numberOfAnswers) {
-          const {
-            id,
-            alias,
-            title,
-            totalTitle: label,
-            qCode,
-            pageType: type,
-          } = page;
-          const summaryIDs = summaryAnswers.map(answer => answer.id);
-          const stripLabel = removeHtml(label) || "";
-          const calculatedSummary = (
-            <Row
-              key={id}
-              id={id}
-              alias={alias}
-              title={title}
-              type={answerMatrix[type]}
-              label={stripLabel}
-              qCode={qCode || ""}
-              summaryAnswers={summaryIDs}
-            />
-          );
-          rowBuilder.push(calculatedSummary);
-        }
+const flattenAnswers = data => {
+  const answers = data.reduce((acc, item) => {
+    const answer = item.answers.map((ans, index) => {
+      if (index > 0) {
+        return {
+          title: item.title,
+          alias: item.alias,
+          nested: true,
+          ...ans,
+        };
+      } else {
+        return {
+          title: item.title,
+          alias: item.alias,
+          ...ans,
+        };
       }
-      return rowBuilder;
     });
+    acc.push(...answer);
+    return acc;
+  }, []);
+  return answers;
+};
 
-    content.push(rows);
-  }
+const handleBlurReducer = ({ type, payload, mutation }) => {
+  const {
+    updateConfirmation,
+    updateCalculatedSummaryPage,
+    updateOption,
+    updateAnswer,
+  } = mutation;
+
+  const mutationVariables = inputValues => {
+    return {
+      variables: {
+        input: {
+          ...inputValues,
+        },
+      },
+    };
+  };
 
-  return content;
+  const { id, qCode } = payload;
+  if (questionMatrix[type] === "Confirmation question") {
```

Comment:
Hi Tom, could you change "Confirmation question" to use the const you've made up above, please?